pkg/quota/v1/evaluator/core/pods.go

```diff
@@ -308,6 +308,8 @@ func podMatchesScopeFunc(selector corev1.ScopedResourceSelectorRequirement, obje
 		return !isBestEffort(pod), nil
 	case corev1.ResourceQuotaScopePriorityClass:
 		return podMatchesSelector(pod, selector)
+	case corev1.ResourceQuotaScopeCrossNamespacePodAffinity:
+		return usesCrossNamespacePodAffinity(pod), nil
```

Comment:
Should we guard this by feature gate? or, it would have been dropped if the feature is disabled?

Reply:
not sure what the best practice here. @liggitt ?

Reply:
If this returns **false** if the feature is disabled, then cross-namespace pods would not be constrained by the quota. Since this code is used by the API server (at pod creation time) and by the controller-manager (at quota reconciliation time), this is the possible skew we could encounter:

* API server with feature on, active controller-manager with feature off
  * quota would be consumed on pod creation, controller-manager would frequently reset consumed quota to 0 (because it wouldn't think it applied to any pods)
* API server with feature off, active controller-manager with feature on
  * quota would not constrain pod creation (because the API server wouldn't think it applied to any pods), controller-manager would frequently recompute consumed quota

Neither of those scenarios prevents creation or deletion of pods that should be allowed, so I would recommend returning false if the feature is disabled.

Will the scheduler also ignore the new namespace selector field if the feature is disabled?